src/npcmove.cpp

```diff
@@ -3047,46 +3051,17 @@ void npc::pick_up_item()
 template <typename T>
 std::list<item> npc_pickup_from_stack( npc &who, T &items )
 {
-    const bool whitelisting = who.has_item_whitelist();
-    auto volume_allowed = who.volume_capacity() - who.volume_carried();
-    auto weight_allowed = who.weight_capacity() - who.weight_carried();
-    auto min_value = whitelisting ? 0 : who.minimum_item_value();
     std::list<item> picked_up;
 
-    for( auto iter = items.begin(); iter != items.end(); ) {
+    auto iter = items.begin();
+    while( iter != items.end() ) {
```

Comment:
You can also do this.
```suggestion
    for( auto iter = items.begin(); iter != items.end(); ) {
```

Reply:
It was like that initially and I failed to notice that it doesn't increment the iterator. I think the `while` is clearer

Reply:
there are numerous places in the codebase that use that style of `for`, and i would advocate toward consistency. i also don't agree that the while is any clearer (or that the for is clearer for that matter)